src/NHibernate/Impl/AbstractQueryImpl.cs

```diff
@@ -701,14 +727,34 @@ public IQuery SetParameterList(string name, IEnumerable vals, IType type)
 			{
 				throw new ArgumentNullException("type","Can't determine the type of parameter-list elements.");
 			}
+#pragma warning disable 618 // Any(taking non generic IEnumerable) is obsolete for turning it internal
 			if(!vals.Any())
+#pragma warning restore 618
 			{
```

Comment:
`SetParameterList(string name, IEnumerable vals, IType type)` is kept, because the element type is already specified as NH type through the third argument, and switching to `IEnumerable<T>` would require to use reflection for many callers, which have a non typed parameters list source possibly mixing different `IEnumerable<T>` values. (Detached query, multi query, ...) Their only common interface is `IEnumerable`. Covariancy to  `IEnumerable<object>` does not help because `IEnumerable<int>` by example can not vary to `object`, it is a `struct`, while `IEnumerable<string>` can.